### PR TITLE
Add support for decimal amounts on Billing resources

### PR DESCRIPTION
**Pull request by remi-stripe**
r? @ob-stripe 
cc @stripe/api-libraries

### CODE REVIEW

File: invoiceitem.go [hunk 1]
```diff
@@ -5,18 +5,19 @@ import "encoding/json"
...
-	Discountable *bool                    `form:"discountable"`
-	Invoice      *string                  `form:"invoice"`
-	Period       *InvoiceItemPeriodParams `form:"period"`
-	Quantity     *int64                   `form:"quantity"`
-	Subscription *string                  `form:"subscription"`
-	TaxRates     []*string                `form:"tax_rates"`
-	UnitAmount   *int64                   `form:"unit_amount"`
+	Params        `form:"*"`
+	Amount        *int64                   `form:"amount"`
+	AmountDecimal *float64                 `form:"amount_decimal,high_precision"`
```
Comment (ob-stripe) [inline invoiceitem.go]: Should be `unit_amount_decimal`.

File: invoiceitem.go [hunk 2]
```diff
@@ -39,24 +40,25 @@ type InvoiceItemListParams struct {
...
-	Metadata     map[string]string `json:"metadata"`
-	Period       *Period           `json:"period"`
-	Plan         *Plan             `json:"plan"`
-	Proration    bool              `json:"proration"`
-	Quantity     int64             `json:"quantity"`
-	Subscription *Subscription     `json:"subscription"`
-	TaxRates     []*TaxRate        `json:"tax_rates"`
-	UnitAmount   int64             `json:"unit_amount"`
+	Amount        int64             `json:"amount"`
+	AmountDecimal float64           `json:"amount_decimal,string"`
```
Comment (ob-stripe) [inline invoiceitem.go]: Same here, should be `unit_amount_decimal`.

File: invoice.go [hunk 3]
```diff
@@ -66,16 +66,17 @@ type InvoiceUpcomingInvoiceItemPeriodParams struct {
...
-	Description  *string                                 `form:"description"`
-	Discountable *bool                                   `form:"discountable"`
-	InvoiceItem  *string                                 `form:"invoiceitem"`
-	Period       *InvoiceUpcomingInvoiceItemPeriodParams `form:"period"`
-	Quantity     *int64                                  `form:"quantity"`
-	Schedule     *string                                 `form:"schedule"`
-	TaxRates     []*string                               `form:"tax_rates"`
-	UnitAmount   *int64                                  `form:"unit_amount"`
+	Amount        *int64                                  `form:"amount"`
+	AmountDecimal *float64                                `form:"amount_decimal,high_precision"`
```
Comment (ob-stripe) [inline invoice.go]: Should be `unit_amount_decimal`.

**Comment by remi-stripe** [comment]
@ob-stripe sorry for the mistakes. Fixed, PTAL

**Pull request merged**